lib/ssh_scan/database/api_db_config.rb

```diff
@@ -0,0 +1,14 @@
+module SSHScan
+  class DatabaseConfig
+    def self.set_database
+      db_path = './lib/ssh_scan/database/database_config.yml'
+      db_opts = YAML.load_file(db_path)
+
+      if db_opts['database']['type'].eql? 'mongodb'
+        return SSHScan::Database::MongoDb.from_config_file(db_path)
+      elsif db_opts['database']['type'].eql? 'sqlite'
+        return SSHScan::Database::SQLite.from_config_file(db_path)
+      end
+    end
```

Comment:
Recommend replacing this file with a from_config method that hangs off of SSHScan::Database.

`db = SSHScan::Database.from_config(db_path)`

The method itself would be defined like so....

    module SSHScan
      class Database
        #rest of class
      
        def self.from_config(config_path)
         #some logic here that picks the DB type/object to pass as options
         SSHScan::Database.new(opts)
        end
      end
    end